### ISSUE TITLE
Comments manipulation

### ISSUE THREAD

**Issue by akira28** [kadamwhite/wordpress-rest-api#152]
I'm trying to create a complete Wordpress experience using this nice Node library, but now I'm stuck on trying to figure out how to retrieve comments for a specific post (I've tried the documented posts().id(XXX).comments() with no luck, it just returns "undefined"), and, more importantly, how to create a comment for an article, that doesn't seem to be documented anywhere. Am I missing something?

**Comment by kadamwhite** [comment]
Probably an error in my documentation—I'll look into it and let you know!

**Comment by kadamwhite** [comment]
@akira28 Thank you for raising this issue; you identified that I completely missed the `/comments` endpoint when I upgraded the client to support the V2 API betas! My apologies for that, it means that querying comments is not currently supported in a robust way. I will attempt to add that functionality back in as soon as we get to a stable place with #145, but for now you can get the comments for a post with this code:
```js
// Note that the "wp/v2" is necessary here, because `.root()` expects a path
// starting at the WP-API root endpoint, e.g. your-site.com/wp-json/*
wp.root( 'wp/v2/comments', true ).param( 'post', XXX ).then(function( comments ) {
	console.log( comments );
});
```
The plan is that this will be simplified to `wp.comments().forPost( XXX )` in short order; because the v2 API made `/comments` a top-level resource that's linked from a specific post, though, you could also get the comments embedded within a request for a post using
```js
wp.posts().id( 968 ).embed().then(function( post ) {
	var comments = post._embedded.replies;
	console.log( comments );
});
```
I am going to leave this issue open to track

- [ ] Ensuring that there is a built-in method for building requests against the comments endpoint
- [ ] Documenting how that endpoint works
- [ ] Documenting how to get the comments for a specific post

**Comment by akira28** [comment]
Fantastic! Thank you very much, I was kinda lost :D

**Comment by kadamwhite** [comment]
@akira28 Took a bit, but I just released [v0.7.0](https://www.npmjs.com/package/wordpress-rest-api) with a new `wp.comments()` endpoint handler. By default it only supports `wp.comments.id( n )` to get a specific comment by ID, and `wp.comments().forPost( n )` to get the comments for a given post; but you have access to `.param` so additional [comments query parameters](v2.wp-api.org/reference/comments/) can be added as needed. I am going to close this issue but if you encounter any problems, or want to take a stab at adding custom param query methods to the comments endpoint handler to make your life easier, please feel free to open a new issue or submit a PR!

Thanks again for calling this omission out.
